### PR TITLE
deps: dump dmn-engine from 1.7.4 to 1.8.0

### PR DESCRIPTION
**Pull request by saig0**
## Description

Dumps the `dmn-engine` from 1.7.4 to [1.8.0](https://github.com/camunda/dmn-scala/releases/tag/1.8.0).

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by saig0** [comment]
@korthout this PR updates the DMN engine to the latest version. All tests are green. :heavy_check_mark: 

No new features but some minor bug fixes. The most relevant change is the update to the latest FEEL engine.

Please approve the PR. It should be part of the upcoming `8.2.0` release. :cookie:

File: dmn/src/test/java/io/camunda/zeebe/dmn/DmnEvaluationTest.java
Comment (korthout) [inline dmn/src/test/java/io/camunda/zeebe/dmn/DmnEvaluationTest.java:54]: I'm assuming this change is due to:
- https://github.com/camunda/dmn-scala/pull/206

❓ Do we need to make other adjustments in Zeebe to use this new behavior, or to improve the error messages in Zeebe for this case?

**Review by korthout — APPROVED** [review]
Thanks @saig0 🚀 

LGTM 👍 

❓ I have a single question. Please consider it before merging. I do not think it blocks merging this pull though.

File: dmn/src/test/java/io/camunda/zeebe/dmn/DmnEvaluationTest.java
Comment (saig0) [reply dmn/src/test/java/io/camunda/zeebe/dmn/DmnEvaluationTest.java:54]: Good hint. :+1: 

I removed the special handling for this bug from the DMN integration. 

Otherwise, we don't need any adjustments.

**Comment by saig0** [comment]
bors merge

**Pull request merged**
